modules/common/src/main/scala/surge/streams/replay/KafkaForeverReplayStrategy.scala

```diff
@@ -169,16 +246,49 @@ class TopicResetActor(brokers: List[String], kafkaTopic: String, config: Config)
   }
 
   override def receive: Receive = ready
+
+  /**
+   * Calculate the TopicReplayProgress that has been made by comparing the current offsets with the preReplay offsets.
+   * @param current
+   *   Map[TopicPartition, OffsetAndMetadata]
+   * @param preReplay
+   *   Map[TopicPartition, OffsetAndMetadata]
+   * @return
+   *   TopicReplayProgress
+   */
+  private def calculateProgress(current: Map[TopicPartition, OffsetAndMetadata], preReplay: Map[TopicPartition, OffsetAndMetadata]): TopicReplayProgress = {
+    // A topic-partition is deemed complete if the current offset is greater than or equal to the preReplay offset.
+    TopicReplayProgress(preReplay.transform((tp, o) => topicPartitionCompleted(current, tp, o)))
+  }
+
+  private def topicPartitionCompleted: (Map[TopicPartition, OffsetAndMetadata], TopicPartition, OffsetAndMetadata) => Boolean =
+    (current: Map[TopicPartition, OffsetAndMetadata], tp: TopicPartition, o: OffsetAndMetadata) => current(tp).offset() >= o.offset()
```

Comment:
In the percent complete calculation here I think rather than a boolean isComplete per partition and doing percent based on that we can get more granular here and guesstimate based on the offsets. As an example, if each partition has an end offset of 100 and our current consumer is at offset 50 everywhere we'd be ~50% done in the replay (maybe not exactly because of cleanup policies on Kafka but a reasonable guess) where the boolean isComplete per partition would show 0% complete